standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -4865,10 +4871,26 @@ private boolean drop_partition_common(RawStore ms, String catName, String db_nam
 
           if (isArchived) {
             assert (archiveParentDir != null);
-            wh.deleteDir(archiveParentDir, true, mustPurge, needsCm);
+            if (writeTruncatedBase) {
+              try {
+                addTruncateBaseFile(archiveParentDir, writeId, archiveParentDir.getFileSystem(getConf()));
+              } catch (Exception e) {
+                throw newMetaException(e);
+              }
+            } else {
+              wh.deleteDir(archiveParentDir, true, mustPurge, needsCm);
+            }
           } else {
             assert (partPath != null);
-            wh.deleteDir(partPath, true, mustPurge, needsCm);
+            if (writeTruncatedBase) {
+              try {
+                addTruncateBaseFile(partPath, writeId, archiveParentDir.getFileSystem(getConf()));
```

Comment:
The metadata file has a type information, I think you should not use truncate type, rather create a drop type.
That info might be useful when you do the cleanup and have to decide whether to delete the whole partition directory or not

Reply:
good idea, will do